fastly/block_fastly_service_v1_logging_datadog_test.go

```diff
@@ -0,0 +1,300 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckServiceV1Exists("fastly_service_v1.foo", &service),
+					testAccCheckFastlyServiceV1DatadogAttributes(&service, []*fst.Datadog{&log1_after_update, &log2}),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "name", name),
+					resource.TestCheckResourceAttr(
+						"fastly_service_v1.foo", "logging_datadog.#", "2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckFastlyServiceV1DatadogAttributes(service *fst.ServiceDetail, datadog []*fst.Datadog) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		conn := testAccProvider.Meta().(*FastlyClient).conn
+		datadogList, err := conn.ListDatadog(&fst.ListDatadogInput{
+			Service: service.ID,
+			Version: service.ActiveVersion.Number,
+		})
+
+		if err != nil {
+			return fmt.Errorf("[ERR] Error looking up Datadog Logging for (%s), version (%d): %s", service.Name, service.ActiveVersion.Number, err)
+		}
+
+		if len(datadogList) != len(datadog) {
+			return fmt.Errorf("Datadog List count mismatch, expected (%d), got (%d)", len(datadog), len(datadogList))
+		}
+
+		log.Printf("[DEBUG] datadogList = %#v\n", datadogList)
+
+		var found int
+		for _, d := range datadog {
+			for _, dl := range datadogList {
+				if d.Name == dl.Name {
+					// we don't know these things ahead of time, so populate them now
+					d.ServiceID = service.ID
+					d.Version = service.ActiveVersion.Number
+					// We don't track these, so clear them out because we also wont know
+					// these ahead of time
+					dl.CreatedAt = nil
+					dl.UpdatedAt = nil
+
+					if !reflect.DeepEqual(d, dl) {
+						return fmt.Errorf("Bad match Datadog logging match,\nexpected:\n(%#v),\ngot:\n(%#v)", d, dl)
+					}
+					found++
+				}
+			}
+		}
+
+		if found != len(datadog) {
+			return fmt.Errorf("Error matching Datadog Logging rules")
+		}
+
+		return nil
+	}
+}
+
+func testAccServiceV1DatadogConfig(name string, domain string) string {
+	return fmt.Sprintf(`
+resource "fastly_service_v1" "foo" {
+  name = "%s"
+
+  domain {
+    name    = "%s"
+    comment = "tf-datadog-logging"
+  }
+
+  backend {
+    address = "aws.amazon.com"
+    name    = "amazon docs"
+  }
+
+  logging_datadog {
+    name   = "datadog-endpoint"
+    token  = "token"
+    region = "US"
+    format = "%%h %%l %%u %%t \"%%r\" %%>s %%b"
+  }
+
+  force_destroy = true
+}
+`, name, domain)
+}
+
+func testAccServiceV1DatadogConfig_update(name, domain string) string {
+	return fmt.Sprintf(`
+resource "fastly_service_v1" "foo" {
+  name = "%s"
+
+  domain {
+    name    = "%s"
+    comment = "tf-datadog-logging"
+  }
+
+  backend {
+    address = "aws.amazon.com"
+    name    = "amazon docs"
+  }
+
+  logging_datadog {
+    name   = "datadog-endpoint"
+    token  = "t0k3n"
+    region = "EU"
+    format = "%%h %%l %%u %%t \"%%r\" %%>s %%b %%T"
+  }
+
+  logging_datadog {
+    name  = "another-datadog-endpoint"
+    token = "another-token"
+  }
```

Comment:
Why doesn't this second logging endpoint have the value included that was set here

https://github.com/terraform-providers/terraform-provider-fastly/pull/242/files#diff-952caa4401e24987ac6e257a290b1f32R161